kvbc/include/categorization/details.h

```diff
@@ -96,5 +97,47 @@ inline void sortAndRemoveDuplicates(std::vector<std::string> &vec) {
   std::sort(vec.begin(), vec.end());
   vec.erase(std::unique(vec.begin(), vec.end()), vec.end());
 }
+// A LocalWriteBatch is used in a multithreaded context, where several jobs need to collect updates.
+// Each job gets its LocalWriteBatch.
+// then all updates are added to the Global NativeWriteBatch, which is not thread safe.
+struct LocalWriteBatch {
+  template <typename KeySpan, typename ValueSpan>
+  void put(const std::string &cFamily, const KeySpan &key, const ValueSpan &value) {
+    puts_.push({cFamily,
+                {std::string(reinterpret_cast<const char *>(key.data()), key.size()),
+                 std::string(reinterpret_cast<const char *>(value.data()), value.size())}});
+    order_.push(put_op_);
+  }
+
+  void moveToBatch(storage::rocksdb::NativeWriteBatch &batch) {
```

Comment:
Nit: I wish std::queue::pop() was noexcept, making this method thread-safe. One way to "achieve" it is to make the whole method `noexcept`, denoting that it either succceeds or we cannot continue safely.